Description: Fixes problems with text2skin skin enigma.
Author: Andreas Brugger <brougs78@gmx.net>
Origin: http://vdrportal.de/board/thread.php?postid=343665#post343665

Index: vdr/osdbase.c
===================================================================
--- vdr.orig/osdbase.c	2015-02-12 18:35:37.923369859 +0100
+++ vdr/osdbase.c	2015-02-12 18:35:37.919369768 +0100
@@ -332,6 +332,7 @@
 
 void cOsdMenu::CursorUp(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int tmpCurrent = current;
   int lastOnScreen = first + displayMenuItems - 1;
   int last = Count() - 1;
@@ -370,6 +371,7 @@
 
 void cOsdMenu::CursorDown(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int tmpCurrent = current;
   int lastOnScreen = first + displayMenuItems - 1;
   int last = Count() - 1;
@@ -410,6 +412,7 @@
 
 void cOsdMenu::PageUp(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int oldCurrent = current;
   int oldFirst = first;
   current -= displayMenuItems;
@@ -444,6 +447,7 @@
 
 void cOsdMenu::PageDown(void)
 {
+  displayMenuItems = displayMenu->MaxItems();
   int oldCurrent = current;
   int oldFirst = first;
   current += displayMenuItems;
