Description: Use ncursesw header file
Author: Tobias Grimm <tg@e-tobi.net>

Index: vdr/PLUGINS/src/skincurses/skincurses.c
===================================================================
--- vdr.orig/PLUGINS/src/skincurses/skincurses.c	2012-03-28 19:57:04.000000000 +0200
+++ vdr/PLUGINS/src/skincurses/skincurses.c	2012-03-28 20:07:38.000000000 +0200
@@ -6,7 +6,7 @@
  * $Id: skincurses.c 2.8 2012/03/11 14:42:52 kls Exp $
  */
 
-#include <ncurses.h>
+#include <ncursesw/ncurses.h>
 #include <vdr/osd.h>
 #include <vdr/plugin.h>
 #include <vdr/skins.h>
