Description: Use ncursesw header file
Author: Tobias Grimm <tg@e-tobi.net>

Index: vdr/PLUGINS/src/skincurses/skincurses.c
===================================================================
--- vdr.orig/PLUGINS/src/skincurses/skincurses.c	2013-03-10 18:35:39.000000000 +0100
+++ vdr/PLUGINS/src/skincurses/skincurses.c	2013-03-10 18:36:53.000000000 +0100
@@ -6,7 +6,7 @@
  * $Id: skincurses.c 2.12 2013/03/03 15:33:59 kls Exp $
  */
 
-#include <ncurses.h>
+#include <ncursesw/ncurses.h>
 #include <vdr/osd.h>
 #include <vdr/plugin.h>
 #include <vdr/skins.h>
