Index: vdr/timers.c
===================================================================
--- vdr.orig/timers.c	2012-06-06 21:30:58.000000000 +0200
+++ vdr/timers.c	2012-06-07 00:27:50.000000000 +0200
@@ -829,7 +829,7 @@
 }
 
 cSortedTimers::cSortedTimers(void)
-:cVector(Timers.Count())
+:cVector<const cTimer *>(Timers.Count())
 {
   for (const cTimer *Timer = Timers.First(); Timer; Timer = Timers.Next(Timer))
       Append(Timer);
