--- timers.c	2013/03/16 10:37:10	2.17
+++ timers.c	2013/03/25 10:44:46
@@ -720,8 +720,10 @@
   for (cTimer *ti = First(); ti; ti = Next(ti)) {
       if (!ti->Recording() && ti->Matches(t)) {
          if (ti->Pending()) {
-            if (ti->Index() > LastPending)
+            if (ti->Index() > LastPending) {
                LastPending = ti->Index();
+               return ti;
+               }
             else
                continue;
             }
